File: app/models/restaurant.rb
# frozen_string_literal: true

# == Schema Information
#
# Table name: restaurants
#
#  id          :bigint           not null, primary key
#  name        :string
#  city_id     :bigint           not null
#  slug        :string
#  website     :string
#  description :text
#  all_vegan   :boolean
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#
class Restaurant < ApplicationRecord
  belongs_to :city

  validates :name, presence: true
  validates :description, presence: true

  def city_name
    city
  end
end

File: src/VegDex.Core/Entities/Restaurant.cs
namespace VegDex.Core.Entities;

public class Restaurant : Entity
{
    public Restaurant() { }
    public bool AllVegan { get; set; }
    public City City { get; set; }
    public int CityId { get; set; }
    public string? Description { get; set; }
    public string Name { get; set; }
    public string Slug { get; set; }
    public string? Website { get; set; }
    public static Restaurant Create(
        int restaurantId,
        string name,
        City city,
        string description = null,
        string website = null,
        bool allVegan = false)
    {
        var restaurant = new Restaurant
        {
            Id = restaurantId,
            Name = name,
            City = city,
            Description = description,
            Website = website,
            AllVegan = allVegan
        };
        return restaurant;
    }
}
